Scoop/Controller/Home/HomeViewController.swift

```diff
@@ -32,13 +38,19 @@ class HomeViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         view.backgroundColor = .white
+
         setupHeaderView()
-        setupTableView()
-        setupPostRideButton()
         setupNotificationButton()
         setupRefreshControl()
+        setupPostRideButton()
         
+        // How to make the thread wait until getRides is finished calling
         getRides()
+        if activeRides.isEmpty && pendingRides.isEmpty {
+            setupEmptyState()
+        } else {
+            setupTableView()
+        }
```

Comment:
wait yes vin, i want it to happen only after get rides is complete but i also dont want to call it inside the get rides function because that function is also called in other places

Reply:
Yea, I don't think there is a way to update your views when your property changes (unless you use SwiftUI). You will have to put this inside of the networking call. Since you are calling `getRides` in multiple places, you could have a separate function for the first time you want to fetch the rides and have this UI update inside. Another way you could do this would be to have just `if activeRides.isEmpty && pendingRides.isEmpty { setupEmptyState() }` inside of the `getRides` networking call (since this empty state should always be triggered if these properties are empty). And depending on what your `setupTableView()` function does, you may not need to actually include this line at all in the network request closure. You would then instead just refresh the table view inside of the networking call. Does this make sense?

Reply:
gotcha, thx vin!!!!!!!

Reply:
Oh yea to add on you could hide/show the table view as well.

Reply:
ok yeah perf i decided to just: 
`if strongSelf.activeRides.isEmpty && strongSelf.pendingRides.isEmpty {
                    strongSelf.tableView.isHidden = true
                    strongSelf.setupEmptyState()
                } else {
                    DispatchQueue.main.async {
                        strongSelf.tableView.reloadData()
                    }
                }`
inside of the networking call and then just calling `setupTableView()` outside in the viewDidLoad

Reply:
oh wow that did not format nicely at all

Reply:
Yea, that's how I would do it. You might need to set `tableView.isHidden = false` in the else in case the user refreshes and ends up getting a ride.

Reply:
ok good point, thx thx again!!!!!!